subsys/bluetooth/controller/hci_driver.c

```diff
@@ -503,6 +511,12 @@ static void event_packet_process(const struct device *dev, uint8_t *hci_buf)
 	struct bt_hci_evt_hdr *hdr = (void *)hci_buf;
 	struct net_buf *evt_buf;
 
+	if (hdr->len + sizeof(*hdr) > HCI_RX_BUF_SIZE) {
+		LOG_ERR("Event buffer too small. %u > %u",
```

Comment:
We should ensure that we don't continue processing an event if the buffer was too small even when logging is disabled.
Either by asserting or immediately discard the received event.

Reply:
k_panic() seems most appropriate. By the time we enter here I think it's already too late